functions/Copy-SqlMaintenancePlan.ps1

```diff
@@ -0,0 +1,237 @@
+
+Windows Authentication will be used if DestinationSqlCredential is not specified. SQL Server does not accept Windows credentials being passed as credentials.
+
+To connect as a different Windows user, run PowerShell as that user.
+
+.NOTES 
+Author: Shawn Melton (@wsmelton)
+Requires: sysadmin access on SQL Servers
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Copy-SqlMaintenancePlan
+
+.EXAMPLE
+Copy-SqlMaintenancePlan -Source sqlserver2014a -Destination sqlcluster
+Copies all maintenance plans from sqlserver2014a to sqlcluster, using Windows credentials. If maintenance plans exist with the same name on sqlcluster, they will be skipped.
+
+.EXAMPLE   
+Copy-SqlMaintenancePlan -Source sqlserver2014a -Destination sqlcluster -Plan MyBackup -SqlCredential $cred -Force
+
+Copies a single maintenance plan, the MyBackup, from sqlserver2014a to sqlcluster, using SQL credentials for sqlserver2014a and Windows credentials for sqlcluster. If a plan with the same name exist on sqlcluster, it will be removed and recreated because -Force was used.
+
+.EXAMPLE   
+Copy-SqlMaintenancePlan -Source sqlserver2014a -Destination sqlcluster -WhatIf -Force
+
+Shows what would happen if the command were executed using -Force.
+#>
+	[CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true)]
+		[object]$Source,
+		[parameter(Mandatory = $true)]
+		[object]$Destination,
+		[System.Management.Automation.PSCredential]$SourceSqlCredential,
+		[System.Management.Automation.PSCredential]$DestinationSqlCredential,
+		[switch]$Force
+	)	
+	DynamicParam { if ($source) { return (Get-ParamSqlMaintenancePlans -SqlServer $source -SqlCredential $SourceSqlCredential) } }
+	
+	BEGIN
+	{
+		$plans = $psboundparameters.MaintenancePlans
+		$exclude = $psboundparameters.Exclude
+
+		$sourceserver = Connect-SqlServer -SqlServer $Source -SqlCredential $SourceSqlCredential
+		$destserver = Connect-SqlServer -SqlServer $Destination -SqlCredential $DestinationSqlCredential
+
+		$source = $sourceserver.DomainInstanceName
+		$destination = $destserver.DomainInstanceName
+	}
+	
+	PROCESS
+	{
+		$qryPlanNames = "SELECT plans.[name] AS plan_name, plans.id AS plan_id
```

Comment:
Please use $sql when writing SQL. This makes SQL queries easier to find in all commands.